src/core/interfaces/executable.ts

```diff
@@ -1,4 +1,5 @@
 // TODO: Why is this an abstract class if it holds no logic ?!
+// Abstract classes can be extended, while interfaces can not. Up for Review? - WBH
```

Comment:
That's the point, why would we extend something that holds no data neither logic, we're using it as an interface their.

Reply:
If we at some point want to have logic in the executable that needs to be extended, this will be critical for not breaking the code.